openqabot/approver.py

```diff
@@ -156,15 +156,61 @@ def validate_job_qam(self, job: int) -> bool:
             return False
         return True
 
+    def _was_older_job_ok(
+        self,
+        failed_job_id: int,
+        inc: int,
+        job: dict,
+        oldest_build_usable: datetime,
+        regex: Pattern[str],
+    ) -> bool:
+        job_build = job["build"][:-2]
+        job_build_date = datetime.strptime(job_build, "%Y%m%d")
+
+        # Check the job is not too old
+        if job_build_date < oldest_build_usable:
+            log.info(
+                "Cannot ignore aggregate failure %s for update %s because: Older jobs are too old to be considered"
+                % (failed_job_id, inc)
+            )
+            return False
+
+        if job["result"] != "passed" and job["result"] != "softfailed":
+            return False
```

Comment:
```suggestion
            return None
```

Reply:
Well, I don't really care much about what this code does. I'm just pointing out that it would behave differently with your change. In case it is acceptable the commit message should be changed to reflect that.

However, I tend to keep the behavior the same because introducing a change nobody has asked for seems very needless. I also don't think the code readability is notably worse with three return values. (There's still the same number of branches.)

Reply:
yes, you are right. That shouldn't be hidden in a "refactoring" commit. I will use your proposed approach and we can keep the true-boolean approach in mind as a potential explicit later approach